source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -0,0 +1,531 @@
+# icat.nix
+{ lib
+, stdenv
+, fetchFromGitHub
+, imlib2
+, xorg
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+
+  src = fetchFromGitHub {
+    owner = "atextor";
+    repo = "icat";
+    rev = "v0.5";
+    sha256 = "0wyy2ksxp95vnh71ybj1bbmqd5ggp13x3mk37pzr99ljs9awy8ka";
+  };
+
+  buildInputs = [ imlib2 xorg.libX11 ];
+}
+```
+
+:::{note}
+Only add the top-level `xorg` derivation to the input attrset, rather than the full `xorg.libX11`, as the latter would cause a syntax error.
+
+Because Nix is lazily-evaluated, including the dependency this way is safe to do and doesn't actually include all of `xorg` into the build context.
+:::
+
+Run the last command again:
+```console
+$ nix-build -A icat
+this derivation will be built:
+  /nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv
+...
+error: builder for '/nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv' failed with exit code 2;
+       last 10 log lines:
+       >   195 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
+       >       |   ^~~~~~~
+       > icat.c: In function 'main':
+       > icat.c:319:33: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
+       >   319 |                                 write(tempfile, &buf, 1);
+       >       |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
+       > gcc -o icat icat.o -lImlib2
+       > installing
+       > install flags: SHELL=/nix/store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15/bin/bash install
+       > make: *** No rule to make target 'install'.  Stop.
+       For full logs, run 'nix log /nix/store/x1d79ld8jxqdla5zw2b47d2sl87mf56k-icat.drv'.
+```
+
+The missing dependency error is solved, but there is now another problem: `make: *** No rule to make target 'install'.  Stop.`
+
+### installPhase
+The `stdenv` is automatically working with the `Makefile` that comes with `icat`: you can see in the console output that `configure` and `make` are executed without issue, so the `icat` binary is compiling successfully.
+
+The failure occurs when the `stdenv` attempts to run `make install`: the `Makefile` included in the project happens to lack an `install` target, and the `README` in the `icat` repository only mentions using `make` to build the tool, leaving the installation step up to users.
+
+To add this step to your derivation, use the [`installPhase` attribute](https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase), which contains a list of command strings to execute to perform the installation.
+
+Because the `make` step completes successfully, the `icat` executable is available in the build directory, and you only need to copy it from there to the output directory.
+
+In Nix, this location is stored in the `$out` variable, accessible in the derivation's component scripts; create a `bin` directory within that and copy the `icat` binary there:
```

Comment:
```suggestion
In Nix, the output directory is stored in the `$out` variable, accessible in the derivation's component scripts.
Create a `bin` directory within the `$out` directory and copy the `icat` binary there:
```

Reply:
I should learn to use fewer semicolons in this writing.